### ISSUE TITLE
"right_arrow" not recognized with "left_shift" AND "left_option"

### ISSUE THREAD

**Issue by cmbail** [pqrs-org/Karabiner-Elements#2574]
Steps to recreate:
- Create Simple modification for device: ThinkPad USB Keyboard with TrackPoint
    - from "left_command" to "left_option"
    - from  "left_option" to "left_command"

Observed behavior in EventViewer:
"left_shift" + "left_option" (physical "left_command") [observed] + "right_arrow" [not observed]
"left_shift" + "right_arrow" [observed]
"left_option" (physical "left_command") + "right_arrow" [observed]
"left_shift" + "left_option" (physical "left_command") + "left_arrow" [observed]

Expected behavior:
"left_shift" + "left_option" (physical "left_command") + "right_arrow" [observed]

**Comment by cmbail** [comment]
Confirmed this is hardware specific. It works as expected with a ThinkPad **Compact** USB Keyboard with TrackPoint.

**Comment by cmbail** [comment]
But it's not a hardware fault. I tried three identical keyboards, they all have the same issue. Possibly a flawed design for this model and they are incapable of outputting Win+Shift+Right in combination. (Even though Win+Shift+Left *is* output correctly.) Or some weird software bug.

The model is Lenovo SK-8855 (FRU 03X8465)

**Comment by jimdigriz** [comment]
@cmbail I think here the compact edition had a firmware update to resolve this:

https://download.lenovo.com/pccbbs/options/tp_compact_usb_kb_with_trackpoint_fw.txt
https://download.lenovo.com/pccbbs/options/tp_compact_usb_kb_with_trackpoint_fw.exe

I have the SK-8855 (USB HID `17ef:6009`) and unable to find an official firmware for it to resolve the same problem. The update `exe` complains with "Can't find HID Device" so I suspect it is looking for the compact (`17ef:6065`).

Might have to do something similar to the efforts on https://github.com/lentinj/tp-compact-keyboard/issues/32 but for the SK-8855.

**Comment by kocane** [comment]
@jimdigriz did you pursue this any further or is it likely impossible to make these changes in the SK-8855 firmware?

**Comment by jimdigriz** [comment]
@kocane problem is this involves some reverse engineering which I am really bad/slow at and do not have the time to sink into this with my other commitments.

Personally, I managed to find a reliable PS/2 USB adapter for my [RT3200](https://sharktastica.co.uk/topics/trackpoint_kbs#RT3200) and so getting the SK-8855 working is no longer important for me.

**Comment by kocane** [comment]
@jimdigriz fair enough. It's a bummer it has to be so complicated, such a simple feature. I can't program worth a damn so I'm useless in this.

I also have a RT3200 with a PS/2 adapter that works alright (altough the button <>\ button does nothing, lol?) but imo it's not as good a keyboard as the SK-8855.

**Comment by zhanghongcq** [comment]
> @cmbail I think here the compact edition had a firmware update to resolve this:
> 
> https://download.lenovo.com/pccbbs/options/tp_compact_usb_kb_with_trackpoint_fw.txt https://download.lenovo.com/pccbbs/options/tp_compact_usb_kb_with_trackpoint_fw.exe
> 
> I have the SK-8855 (USB HID `17ef:6009`) and unable to find an official firmware for it to resolve the same problem. The update `exe` complains with "Can't find HID Device" so I suspect it is looking for the compact (`17ef:6065`).
> 
> Might have to do something similar to the efforts on [lentinj/tp-compact-keyboard#32](https://github.com/lentinj/tp-compact-keyboard/issues/32) but for the SK-8855.

It need modify the PID & VID at the position from 794b0H to 9747H by the actual PID&VID, Attention: Need XOR key 0x5A.
![image](https://github.com/pqrs-org/Karabiner-Elements/assets/25217525/97f2f62b-3439-4f32-bbc9-30fa3bec4579)

**Comment by cmbail** [comment]
@zhanghongcq very helpful, thank you. I resorted to using only wireless variants of Thinkpad keyboards with my Mac, this may unlock my USB ones.
